app/src/main/java/org/gdg/frisbee/android/common/GdgNavDrawerActivity.java

```diff
@@ -178,20 +184,36 @@ private void onDrawerItemClick(final MenuItem item) {
                 }
                 break;
             case Const.DRAWER_HOME:
-                navigateTo(MainActivity.class, data);
+                if (!(this instanceof MainActivity)) {
+                    navigateTo(MainActivity.class, data);
+                }
                 break;
             case Const.DRAWER_GDE:
-                navigateTo(GdeActivity.class, data);
+                if (!(this instanceof GdeActivity)) {
+                    navigateTo(GdeActivity.class, data);
```

Comment:
Can you expand how would the check look like in the navigateTo? The only way I can think of is:

```java

   if (this.getClass() == activityClass) {
            
   }

```

Which would probably work. I'll test it!